### ISSUE TITLE
Project goals

### ISSUE THREAD

**Issue by jabrena** [mob41/jEv3dev#1]
Hi @mob41,

Why do you prefer to teach programming with a visual approach?

Cheers

Juan Antonio

**Comment by a1aw** [comment]
Hi jabrena,

In fact, I do not prefer learning programming in a visual way, those blocks, in a lower-level computing, the code(s) are quite messy and having low performance.

However, having a friendly and visual blocks to program for beginners in ev3dev would be more program-friendly for them. What I want to do is to do something like "Blocks to Code", letting the beginners to understand the blocks in a "code" format.

You want to contribute in this project? Though I don't have time currently.

Best regards,
Anthony

**Comment by jabrena** [comment]
Hi @mob41,

No, it was simply curiosity @mob41. In my case, I am working on a Textual environment to teach Java, so the opposite approach. :)

Anyway, I checked your repo about `EV3Dev-lang-java` and I liked some classes. If you like, we could collaborate in the future.

Cheers

Juan Antonio

**Comment by a1aw** [comment]
Seems the issue 788 in ev3dev's repository about the ev3dev IDE still does not have a clear idea about having, or how to have an ev3dev IDE for beginners. We may continue developing this.

By the way, I have to finish my osumer2 project before picking up this project up. You can take a look of the code and see what to add then 😄

**Comment by a1aw** [comment]
Blocks and Graphics is broken currently. I think I need to make a rewrite of it. I am still thinking to use Rails (Just like in NXT GUI) or Lines (Just like in EV3 GUI), or just flow charts...

Using rails and flow charts would be more easy to develop because it is straight forward to do so, but lines will be a bit tricky to handle.

**Comment by jabrena** [comment]
Hi @mob41,

I will explain the point of view:

we invest many time on Java, a Textual programming language, so why invest time in the development of a visual environment to explain a textual programming language? I think that it is better if we help the new developer with Java using Java not another visual solution. In my opinion, visual environment is not the solution to teach Java... (Humble opinion)

In the other hand, I liked your `LCD` support. I will add some implementation about `LCD` using your approach in the next release. Congrats!!!

**Example:**
https://www.youtube.com/watch?v=Gxew3aNH6ks

Cheers

Juan Antonio

**Comment by jabrena** [comment]
This summer, I will work with `ROS` if you like, we could do some nodes :)

**Comment by a1aw** [comment]
Wait, you are copying and pasting my code directly?

https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/develop/src/main/java/ev3dev/actuators/ev3/lcd/EV3GraphicsLCD.java

The code is almost the same as my version.

I won't recommend copying this as the code is quite dirty. Especially if running on the EV3, Graphics processing takes such much time, where the ```getRGB()``` method should be changed.

**Comment by jabrena** [comment]
Yes, I have to do a revision.  At the moment the class has a lot of methods in compare with the interface but I liked the idea. I agree with you, the method getRGB() is not the best method for a monochrome environment.
![](https://github.com/ev3dev-lang-java/ev3dev-lang-java/blob/develop/docs/uml/graph.png?raw=true)

The issue to review is:
https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/259
